test/relations.test.js

```diff
@@ -709,40 +710,142 @@ describe('relations', function() {
       }
     });
 
-    it('should fetch scoped instances with paging filters', function(done) {
-      Physician.create(function(err, physician) {
-        physician.patients.create({name: 'a'}, function() {
-          physician.patients.create({name: 'z'}, function() {
-            physician.patients.create({name: 'c'}, function() {
-              verify(physician);
+    describe('fetch scoped instances with paging filters', function() {
+      var samplePatientId;
+      var physician;
+
+      beforeEach(createSampleData);
+
+      context('with filter skip', function() {
+        it('skips the first patient', function(done) {
+          physician.patients({skip: 1}, function(err, ch) {
+            should.not.exist(err);
+            should.exist(ch);
+            ch.should.have.lengthOf(2);
+            ch[0].name.should.eql('z');
```

Comment:
@b-admike there are some PRs in master that fix the order problem after my PR checked in, good point, I think I should backport them together.